### PR TITLE
statik 0.10.4 (new formula)

### PR DESCRIPTION
**Pull request by thanethomson**
This commit adds Statik (https://getstatik.com) as a new formula for
Homebrew. Statik is a Python-based static web site generator aimed at
developers, and allows for the creation of almost any kind of static web
site (not at all limited to blogs). See the wiki for more details:
https://github.com/thanethomson/statik/wiki

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/statik.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    url "https://pypi.python.org/packages/ca/ca/c2436fdb7bb75d772d9fa17ba60c4cfded6284eed053a7274b2beb96596a/SQLAlchemy-1.1.4.tar.gz"
+    sha256 "701b57d628b9fa1cfb82f10665e7214d5d2db23251ca6f23b91c5f56fcdbdeb5"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system bin/"statik", "--version"
```
Comment (MikeMcQuaid) [inline Formula/statik.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by thanethomson** [comment]
I've now added a more detailed unit test for installation - it tests the actual functionality of Statik.

File: Formula/statik.rb (hunk 1 above)
Comment (thanethomson) [inline Formula/statik.rb]: Done 😀

File: Formula/statik.rb [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    (testpath/"config.yml").write <<-EOS.undent
+      project-name: Homebrew Test
+      base-path: /
+    EOS
+    (testpath/"models/Post.yml").write <<-EOS.undent
```
Comment (MikeMcQuaid) [inline Formula/statik.rb]: The ones on a single line you can probably just do `.write("title: String")`?

File: Formula/statik.rb [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    EOS
+    (testpath/"templates/post.html").write <<-EOS.undent
+      <html>
+      <head><title>Post</title></head>
+      <body>{{ post.title }}</body>
+      </html>
+    EOS
+    system bin/"statik"
+
+    assert File.exist? testpath/"public/index.html"
```
Comment (MikeMcQuaid) [inline Formula/statik.rb]: Add a second argument to the asserts to describe what the failing assertion means e.g. `, "index.html was not created!"`

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
Almost there, some tiny nits remaining.

File: Formula/statik.rb (hunk 2 above)
Comment (thanethomson) [inline Formula/statik.rb]: Done

File: Formula/statik.rb (hunk 3 above)
Comment (thanethomson) [inline Formula/statik.rb]: And done 😀

**Comment by thanethomson** [comment]
Changes made 😀

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**